### PR TITLE
camelCasing all manifest parameters

### LINKED ISSUE #594: camelCase for all manifest parameters
There is some inconsistency in naming of manifest parameters. Most are written in camelCase, except:

- init_containers
- pod_priority_class_name

For the latter one even the reference list it as `podPriorityClassName`. I'd suggest to change the parameters to camelCase style and use snake_case only for configuration parameters.

Thanks @knan-nrk for pointing via Slack.

### PR DESCRIPTION
**Pull request by FxKu**
init_containers -> initConatiners
pod_priority_class_name -> podPriorityClassName

backwards compatibility is treated in unmarshalling method and with extra fields

fixes #594

### CODE REVIEW

File: pkg/apis/acid.zalan.do/v1/util_test.go [hunk 1]
```diff
@@ -170,6 +170,7 @@ var unmarshalCluster = []struct {
 		marshal: []byte(`{"kind":"Postgresql","apiVersion":"acid.zalan.do/v1","metadata":{"name":"acid-testcluster1","creationTimestamp":null},"spec":{"postgresql":{"version":"","parameters":null},"volume":{"size":"","storageClass":""},"patroni":{"initdb":null,"pg_hba":null,"ttl":0,"loop_wait":0,"retry_timeout":0,"maximum_lag_on_failover":0,"slots":null},"resources":{"requests":{"cpu":"","memory":""},"limits":{"cpu":"","memory":""}},"teamId":"","allowedSourceRanges":null,"numberOfInstances":0,"users":null,"clone":{}},"status":{"PostgresClusterStatus":"Invalid"}}`),
 		err:     nil},
 	// example with detailed input manifest
+	// and dprecated pod_priority_class_name -> podPriorityClassName
```
Comment (sdudoladov) [inline pkg/apis/acid.zalan.do/v1/util_test.go]: dEprecated

**Comment by sdudoladov** [comment]
:+1:

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Comment by erthalion** [comment]
:+1:

**Pull request merged**